business/istio_config.go

```diff
@@ -45,14 +45,77 @@ type IstioConfigCriteria struct {
 	IncludeServiceRoleBindings    bool
 	IncludeSidecars               bool
 	IncludeAuthorizationPolicies  bool
-	IncludePeerAuthentication     bool
+	IncludePeerAuthentications    bool
 	IncludeWorkloadEntries        bool
 	IncludeRequestAuthentications bool
 	IncludeEnvoyFilters           bool
 	IncludeAttributeManifests     bool
 	IncludeHttpApiSpecBindings    bool
 	IncludeHttpApiSpecs           bool
 	LabelSelector                 string
+	WorkloadSelector              string
+}
+
+func (icc IstioConfigCriteria) Include(resource string) bool {
+	// Flag used to skip object that are not used in a query when a WorkloadSelector is present
+	isWorkloadSelector := icc.WorkloadSelector != ""
+	switch resource {
```

Comment:
Similarly to the previous comment in the UI PR review. Do you think there is a way to reduce the switch by using the `reflect` or dynamic access methods? Probably this is because of the architecture of the package so it might change after the already commented refactor.

Reply:
I don't think it can be changed without migrating more things in the model.

Basically the IstioConfigCriteria is taking the flags of which types to use because we have a single istio endpoint to fetch all types with a single IstioConfig abstraction.

Also this Criteria is used to decide what to check or not.

Point is that now we have another scenario, IstioConfig which is related/linked with a Workload, so that logic is done on this helper method.

What I don't want is to move this ParseCriteria to typeless, i.e. just storing a "type string" and checking if included or not, I'm less sure, I prefer to keep this.

At some point this package will be revisit, there are deprecated types that will be removed and then we can think more movements, but I think having an Include() and not propagating this complexity beyond the helper is a good idea.

Reply:
Yes, I was a bit unsure of this one. I guess that when we revisit this package we will be able to get rid of some snippets like this one.